src/messages/Modal.ts

```diff
@@ -0,0 +1,18 @@
+import { constant, Decoder, mixed } from 'decoders';
+import { labeledDecoder, LabeledMsg } from './LabeledMsg';
+
+/**
+ * A message used to publish a generic messages
+ * between the clients and the host application.
+ */
+export interface LabeledModal extends LabeledMsg<'modalRequest', any> {
```

Comment:
This file should be called `ModalRequest` and should follow the same pattern as other messages, like `NavRequest`. That means this interface would be come `LabeledModalRequest` and should extend `LabeledMsg<'modalRequest', ModalRequest>`, where `ModalRequest` is a type defined and exported in this file that contains all data involved in requesting a modal from the host application.

Additionally, you'll need to add the `LabeledModalRequest` to the `ClientToHost` type in `/messages/ClientToHost.ts`.

Reply:
Changed as requested.